src/util/unesc.js

```diff
@@ -1,19 +1,96 @@
-const whitespace = '[\\x20\\t\\r\\n\\f]';
-const unescapeRegExp = new RegExp('\\\\([\\da-f]{1,6}' + whitespace + '?|(' + whitespace + ')|.)', 'ig');
+// Many thanks for this post which made this migration much easier.
+// https://mathiasbynens.be/notes/css-escapes
+
+const CSS_SPECIAL_CHARS = new Set([
+    "!", "\"", "#", "$", "%", "&", "'", "(", ")", "*", "+", ",", "-", ".", "/", ";", "<", "=", ">", "?", 
+    "@", "[", "]", "^", "`", "{", "|", "}", "~", "_", ":",
+]);
+
+const CSS_SPECIAL_CHARS_HEX_LONG = {}; 
+const CSS_SPECIAL_CHARS_HEX_SHORT = {};
```

Comment:
These seem to be used as maps, so we could use actual `Map`s. It might even speed up this patch even further, since during the lookups (e.g. line 26) we’d then be avoiding prototype chain walks for any misses.